R/intervals_support_funs.R

```diff
@@ -0,0 +1,363 @@
+    warning(paste0("No intervals found with the specified target parameters,",
+                   " groups, and/or after-change needed. No changes made."))
+    return(data[, !names(data) %in% index_colname])
+
+    # If target intervals are found...
+  } else {
+    # The new imputation should not be used non-target parameters
+    new_intervals[, setdiff(param_cols, target_params)] <- FALSE
+
+    # Index the new intervals to be after the original ones
+    new_intervals[["impute"]] <- add_impute_method(new_intervals[["impute"]], target_impute, after)
+    new_intervals[[index_colname]] <- new_intervals[[index_colname]] + 0.5
+  }
+
+  # Remove the target parameters calculation from the original target intervals
+  data[target_rows, target_params] <- FALSE
+
+  # Combine the new and original intervals
+  data <- rbind(data, new_intervals)
+
+  # Filter rows where all row values for param_cols are NA or FALSE
+  param_data <- data[, param_cols, drop = FALSE]
+  rows_no_params <- rowSums(!is.na(replace(param_data, param_data == FALSE, NA))) == 0
+  data <- data[!rows_no_params, , drop = FALSE]
+
+  # Order the intervals by the index column and then remove it
+  data <- data[order(data[[index_colname]]), ]
+  rownames(data) <- seq_len(nrow(data))
+  data[, !names(data) %in% index_colname]
+}
+
+#' Remove specified imputation from the intervals in a PKNCAdata or data.frame (intervals) object.
+#'
+#' @inheritParams interval_add_impute
+#' @param target_impute A character string specifying the imputation method to remove.
+#' @param ... arguments passed to `interval_remove_impute`.
+#' @return A modified object with the specified imputations removed from the targeted intervals.
+#' @examples
+#' d_conc <- data.frame(
+#'   conc = c(1, 0.6, 0.2, 0.1, 0.9, 0.4, 1.2, 0.8, 0.3, 0.2, 1.1, 0.5),
+#'   time = rep(0:5, 2),
+#'   ID = rep(1:2, each = 6),
+#'   analyte = rep(c("Analyte1", "Analyte2"), each = 6)
+#' )
+#'
+#' d_dose <- data.frame(
+#'   dose = c(100, 200),
+#'   time = c(0, 0),
+#'   ID = c(1, 2)
+#' )
+#'
+#' library(PKNCA)
+#' o_conc <- PKNCA::PKNCAconc(d_conc, conc ~ time | ID / analyte)
+#' o_dose <- PKNCA::PKNCAdose(d_dose, dose ~ time | ID)
+#'
+#' intervals <- data.frame(
+#'   start = c(0, 0, 0),
+#'   end = c(3, 5, Inf),
+#'   half.life = c(TRUE, FALSE, TRUE),
+#'   cmax = c(TRUE, TRUE, TRUE),
+#'   impute = c("start_conc0,start_predose", "start_predose", "start_conc0"),
+#'   analyte = c("Analyte1", "Analyte2", "Analyte1")
+#' )
+#'
+#' o_data <- PKNCAdata(o_conc, o_dose, intervals = intervals)
+#'
+#' # Apply interval_remove_impute function
+#' o_data <- interval_remove_impute(data = o_data,
+#'                                  target_impute = "start_conc0",
+#'                                  target_params = "half.life",
+#'                                  target_groups = data.frame(analyte = "Analyte1"))
+#' @export
+interval_remove_impute <- function(data, ...) {
+  UseMethod("interval_remove_impute", data)
+}
+
+#' @export
```

Comment:
issue: No need to export this function, I would use `@keywords internal` instead.
```suggestion
#' @keywords internal
```